src/views/Project/Details/Frameworks/FrameworkDetail/EditFrameworkModal/FrameworkUsersTable/AddFromSearch/index.js

```diff
@@ -0,0 +1,236 @@
+                message: _ts('project.framework.edit', 'membershipListFetchFailure'),
+                duration: notify.duration.SLOW,
+            });
+        },
+        onFatal: () => {
+            notify.send({
+                title: _ts('project.framework.edit', 'afPatch'),
+                type: notify.type.ERROR,
+                message: _ts('project.framework.edit', 'membershipListFetchFatal'),
+                duration: notify.duration.SLOW,
+            });
+        },
+        schemaName: 'usersSearchGetResponse',
+    },
+    userAddRequest: {
+        url: '/framework-memberships/',
+        method: requestMethods.POST,
+        body: ({ params }) => params.membership,
+        onSuccess: ({
+            response,
+            params: {
+                handleUserAdd,
+            },
+            props: {
+                onAddUser,
+            },
+        }) => {
+            onAddUser(response);
+            handleUserAdd(response.member);
+        },
+        onFailure: () => {
+            notify.send({
+                title: _ts('project.framework.edit', 'afPatch'),
+                type: notify.type.ERROR,
+                message: _ts('project.framework.edit', 'membershipAddFailure'),
+                duration: notify.duration.SLOW,
+            });
+        },
+        onFatal: () => {
+            notify.send({
+                title: _ts('project.framework.edit', 'afPatch'),
+                type: notify.type.ERROR,
+                message: _ts('project.framework.edit', 'membershipAddFatal'),
+                duration: notify.duration.SLOW,
+            });
+        },
+        schemaName: 'frameworkMembership',
+    },
+};
+
+const listKeySelector = l => l.id;
+
+@RequestClient(requests)
+export default class AddFrameworkUserFromSearch extends React.PureComponent {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            users: [],
+        };
+
+        this.props.setDefaultRequestParams({
+            handleUsersPull: this.handleUsersPull,
+        });
+    }
+
+    handleUsersPull = (users) => {
+        this.setState({ users });
+    }
+
+    handleUserAdd = (memberId) => {
+        const { users } = this.state;
+        const newUsers = produce(users, (safeUsers) => {
+            const index = users.findIndex(u => u.id === memberId);
+            safeUsers.splice(index, 1);
+        });
+        this.setState({ users: newUsers });
+    }
+
+    listRendererParams = (_, data) => {
+        const {
+            userAddRequest: {
+                pending,
+            } = {},
+        } = this.props;
+        const {
+            selectedUser,
+        } = this.state;
+        const {
+            id: userId,
+        } = data;
+
+        return ({
+            ...data,
+            pending: pending && selectedUser === userId,
+            onAddButtonClick: () => this.handleAddClick(userId),
```

Comment:
Creating a method on render beats the purpose of creating a rendererParams method